sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/EventProcessorTests.MainProcessingLoop.cs

```diff
@@ -581,6 +582,14 @@ public async Task BackgroundProcessingStopsProcessingAllPartitionsWhenShutdown()
             var mockConnection = new Mock<EventHubConnection>();
             var mockProcessor = new Mock<EventProcessor<EventProcessorPartition>>(65, "consumerGroup", "namespace", "eventHub", Mock.Of<TokenCredential>(), options, mockLoadBalancer.Object) { CallBase = true };
 
+            mockLogger
+                .Setup(log => log.EventProcessorPartitionProcessingStopComplete(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>()))
+                .Callback(() =>
+                {
+                    stopCompletionSources[stopCompletionIndex].TrySetResult(true);
```

Comment:
Does this need to be thread-safe? It appears that this test waits for all these stop completion sources to signal completion.. but I was wondering if two threads signaled the same index twice, if one left over one may be false still... which could cause it to hang?

Reply:
I... did not consider that, but yeah.  That's a damn good observation and could very well explain the behavior.